### PR TITLE
[MIR-970] Redesign AccessKeys

### PR DESCRIPTION
**Pull request by golsch**
[Link to jira](https://mycore.atlassian.net/browse/MIR-970).

### CODE REVIEW

**Comment by MyCoRe-Travis** [comment]
Something gone wrong with f0e21e324a711b4a878d5383e527d6bab06584ae! 
 Please Check https://github.com/MyCoRe-Travis/MIR_test_artifacts/tree/3350-pull_request-593

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 1]
```diff
@@ -0,0 +1,244 @@
+const RSC_URL = webApplicationBaseURL + "rsc/accesskey/";
+let labels = undefined;
+
+function getParameterByName(name, url = window.location.href) {
+    name = name.replace(/[\[\]]/g, '\\$&');
+    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+    results = regex.exec(url);
+    if (!results) return null;
+    if (!results[2]) return '';
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:9]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
    if (!results[2]) return "";
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+	var keylistspec = "!@#_%$";
+	var temp = '';
+	var len = plength / 2;
+	var len = len - 1;
+	var lenspec = plength - len - len;
+
+	for (i = 0; i < len; i++)
+		temp += keylistalpha.charAt(Math.floor(Math.random() * keylistalpha.length));
+
+	for (i = 0; i < lenspec; i++)
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:25]: Codacy has a fix for the issue: [Expected { after 'for' condition.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
	for (i = 0; i < lenspec; i++)
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+function generateKey(plength) {
+	var keylistalpha = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ";
+	var keylistint = "123456789";
+	var keylistspec = "!@#_%$";
+	var temp = '';
+	var len = plength / 2;
+	var len = len - 1;
+	var lenspec = plength - len - len;
+
+	for (i = 0; i < len; i++)
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:22]: Codacy has a fix for the issue: [Expected { after 'for' condition.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
	for (i = 0; i < len; i++)
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js (hunk 1 above)
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:9]: Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
    if (!results[2]) {return '';}
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 4]
```diff
@@ -0,0 +1,244 @@
+const RSC_URL = webApplicationBaseURL + "rsc/accesskey/";
+let labels = undefined;
+
+function getParameterByName(name, url = window.location.href) {
+    name = name.replace(/[\[\]]/g, '\\$&');
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:5]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
    name = name.replace(/[\[\]]/g, "\\$&");
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 5]
```diff
@@ -0,0 +1,244 @@
...
+    var regex = new RegExp('[?&]' + name + '(=([^&#]*)|&|#|$)'),
+    results = regex.exec(url);
+    if (!results) return null;
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:8]: Codacy has a fix for the issue: [Expected { after 'if' condition.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
    if (!results) {return null;}
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 6]
```diff
@@ -0,0 +1,244 @@
...
+    $('#accessKeyDelete').prop("disabled", true);
+    $('#accessKeyNew').prop("disabled", true);
+    $('.closeModal').prop("disabled", true);
+}
+
+function enableButtons() {
+    $('#accessKeyUpdate').prop("disabled", false);
+    $('#accessKeyDelete').prop("disabled", false);
+    $('#accessKeyNew').prop("disabled", false);
+    $('.closeModal').prop("disabled", false);
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:88]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
    $(".closeModal").prop("disabled", false);
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 7]
```diff
@@ -0,0 +1,244 @@
...
+
+	for (i = 0; i < lenspec; i++)
+		temp += keylistspec.charAt(Math.floor(Math.random() * keylistspec.length));
+
+	for (i = 0; i < len; i++)
+		temp += keylistint.charAt(Math.floor(Math.random() * keylistint.length));
+
+	temp = temp.split('').sort(function() {
+		return 0.5 - Math.random()
+	}).join('');
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:33]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
	}).join("");
```

File: mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js [hunk 8]
```diff
@@ -0,0 +1,244 @@
...
+
+function disableButtons() {
+    $('#accessKeyUpdate').prop("disabled", true);
+    $('#accessKeyDelete').prop("disabled", true);
+    $('#accessKeyNew').prop("disabled", true);
+    $('.closeModal').prop("disabled", true);
+}
+
+function enableButtons() {
+    $('#accessKeyUpdate').prop("disabled", false);
```
Comment (mcradmin) [inline mir-module/src/main/resources/META-INF/resources/js/mir/accesskey.js:85]: Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/MyCoRe-Org/mir/pullRequest?prid=7162253)
```suggestion
    $("#accessKeyUpdate").prop("disabled", false);
```

**Pull request closed without merging**